### PR TITLE
Rename `type` to `ty` and `_tag` to `type` & remove enums

### LINKED ISSUE #93: Change discriminant from `_tag` to `type`
`type` isn't a valid Rust identifier (unless escaped via `r#type`), so this minimizes chance of collision.

### LINKED ISSUE #97: Move from enums to string unions

### PR DESCRIPTION
**Pull request by tjjfvi**
Fixes #93
Fixes #97

### CODE REVIEW

File: examples/transfer.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const metadata = C.M.fromPrefixedHex(metadataRaw.result);
 const deriveCodec = C.M.DeriveCodec(metadata);
 
 const dest = new C.MultiAddress(
-  C.MultiAddressKind.Id,
+  "Id",
```
Comment (harrysolovay) [inline examples/transfer.ts:21]: So much better

File: frame_metadata/Codec.ts [hunk 2]
```diff
@@ -130,32 +112,33 @@ export const DeriveCodec = (metadata: M.Metadata): DeriveCodec => {
...
     },
-    [M.TypeKind.Tuple]: (ty) => {
+    Tuple: (ty) => {
       return tuple(ty.fields);
     },
-    [M.TypeKind.Primitive]: (ty) => {
-      return primitiveCodecByDiscriminant[ty.kind]!;
+    Primitive: (ty) => {
+      if (ty.kind === "char") return $.str;
+      return $[ty.kind];
```
Comment (harrysolovay) [inline frame_metadata/Codec.ts:136]: YES

File: frame_metadata/TypeVisitor.ts [hunk 3]
```diff
@@ -1,21 +1,21 @@
 import * as U from "../util/mod.ts";
 import * as M from "./Metadata.ts";
 
-export type UnknownByTypeDefKind = { [Tag in M.TypeKind]: unknown };
+export type UnknownByTypeDefKind = { [Tag in M.TyType]: unknown };
 
 export type TypeVisitor<
-  Tag extends M.TypeKind,
+  Tag extends M.TyType,
```
Comment (harrysolovay) [inline frame_metadata/TypeVisitor.ts:7]: Should the name of the type param also change? It is technically still a tag.

Comment (harrysolovay) [reply frame_metadata/TypeVisitor.ts:7]: ... still, might be better as `TyType`

Comment (tjjfvi) [reply frame_metadata/TypeVisitor.ts:7]: Yep, it should

**Review by harrysolovay — APPROVED** [review]

**Pull request merged**